### PR TITLE
Feature/pdct 1274 update the api search method to apply corpus filtering

### PR DESCRIPTION
**Pull request by katybaulch**
# Description

Update search to apply corpus filtering & tests.

Corpus filtering logic explained:

- First corpora validation is app token against DB. At least one of the app token corpora IDs must be present in the DB to continue the search request.
- If the search request IDs are null, we want to search using the app token corpora.
- For the second validation, search request corpora Ids are validated against the app token corpora IDs if the search request param 'corpus_import_ids' is not None. corpus_import_ids must be a subset of app token IDs.

## Proposed version

Please select the option below that is most relevant from the list below. This
will be used to generate the next tag version name during auto-tagging.

- [ ] Skip auto-tagging
- [ ] Patch
- [x] Minor version
- [ ] Major version

Visit the [Semver website](https://semver.org/#summary) to understand the
difference between `MAJOR`, `MINOR`, and `PATCH` versions.

Notes:

- If none of these options are selected, auto-tagging will fail
- Where multiple options are selected, the most senior option ticked will be
  used -- e.g. Major > Minor > Patch
- If you are selecting the version in the list above using the textbox, make
  sure your selected option is marked `[x]` with no spaces in between the
  brackets and the `x`

## Type of change

Please select the option(s) below that are most relevant:

- [ ] Bug fix
- [ ] New feature
- [ ] Breaking change
- [ ] GitHub workflow update
- [ ] Documentation update
- [ ] Refactor legacy code
- [ ] Dependency update

## How Has This Been Tested?

Please describe the tests that you added to verify your changes.

## Reviewer Checklist

- [ ] **DB_CLIENT DEPENDENCY IS ON THE LATEST VERSION**
- [ ] The PR represents a single feature (small driveby fixes are also ok)
- [ ] The PR includes tests that are sufficient for the level of risk
- [ ] The code is sufficiently commented, particularly in hard-to-understand areas
- [ ] Any required documentation updates have been made
- [ ] Any TODOs added are captured in future tickets
- [ ] No FIXMEs remain

### CODE REVIEW

File: app/db/crud/helpers/validate.py [hunk 1]
```diff
@@ -30,3 +30,23 @@ def verify_any_corpora_ids_in_db(db: Session, corpora_ids: list[str]) -> bool:
...
+    :param set[str] corpora_ids: The corpus import IDs we want to
+        validate.
+    :param set[str] valid_corpora_ids: The corpus import IDs
+        we want to validate against.
+    :return bool: Return whether or not all the corpora are valid.
+    """
+    validate_success = corpora_ids.issubset(valid_corpora_ids)
+    if not validate_success:
+        invalid_corpora = set(corpora_ids).difference(valid_corpora_ids)
+        if invalid_corpora != set():
```
Comment (diversemix) [inline app/db/crud/helpers/validate.py:47]: Think this check is redundant give line 44 .. How can this line evaluate to `False` ?

**Review by diversemix — APPROVED** [review]
lgtm - just confirm the `if` statement is necessary

File: app/db/crud/helpers/validate.py (hunk 1 above)
Comment (katybaulch) [reply app/db/crud/helpers/validate.py:47]: Good spot 👓

Comment (katybaulch) [reply app/db/crud/helpers/validate.py]: Updated

**Pull request merged**